optuna/_hypervolume/hssp.py

```diff
@@ -44,10 +44,13 @@ def _solve_hssp_on_unique_loss_vals(
     subset_size: int,
     reference_point: np.ndarray,
 ) -> np.ndarray:
-    assert not np.any(reference_point - rank_i_loss_vals <= 0)
+    if not np.isfinite(reference_point).all():
+        return rank_i_indices[list(range(subset_size))]
+    with np.errstate(invalid="ignore"):
```

Comment:
Do we still need it after we added the guard? Shouldn't the warning happen at all because `inf - inf` will not happen?